src/network/wire/lib.zig

```diff
@@ -102,6 +102,14 @@ pub fn receiveMessage(
     const payload_len = try r.readInt(u32, .little);
     const checksum = try r.readBytesNoEof(4);
 
+    // Calculate total message size
+    const precomputed_total_size = 24; // network (4 bytes) + command (12 bytes) + payload size (4 bytes) + checksum (4 bytes)
+    const total_message_size = precomputed_total_size + payload_len;
+
+    if (total_message_size > MAX_SIZE) {
+        return error.InvaliPayloadLen;
+    }
```

Comment:
Can you factorize this inside a function? For both read and write. That takes `payload_len` as argument and return `!void`

Reply:
And `const MAX_SIZE: usize = 0x02000000; // 32 MB` can be inside this function scope

Reply:
Okay boss will do this now

Reply:
@tdelabro I made send and receive message return !void and if the payload size is too large it returns InvalidPayloadLen

Reply:
@tdelabro also just fixed the typo from InvaliPayloadLen to InvalidPayloadLen